src/infrastructure/content/NetworkMessageHandler.java

```diff
@@ -0,0 +1,79 @@
+package infrastructure.content;
+
+import networking.INotificationHandler;
+import org.json.JSONObject;
+import java.util.HashMap;
+
+/*
+ * NetworkMessageHandler class implements the INotificationHandler interface.
+ * this includes one method i.e @onMessageReceived()
+ * this class is used to notify and to send data to the UI module 
+ * as new user joined or as a user exit or as message sent by a user
+ * 
+ * @author Talha Yaseen (111701019)
+ */
+public class NetworkMessageHandler implements INotificationHandler {
+	
+	/*
+	 * Creating instance of ContentCommunicator class to access map handler and image map
+	 */
+	private ContentCommunicator cc = new ContentCommunicator();
+	
```

Comment:
contentCommunicator cannot be instantiated directly. There is a getContentCommunicator() function. Talk with badal for singleton pattern he is using.